### PR TITLE
Add support for amp-soundcloud

### PR DESCRIPTION
**Pull request by amedina**
Getting this PR started for review and discussion.

### CODE REVIEW

File: includes/embeds/class-amp-soundcloud-embed.php [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+				'height' => $this->args['height'],
+			)
+		);
+	}
+
+	private function get_track_id_from_url( $url ) {
+		$parsed_url = parse_url( $url );
+		parse_str( $parsed_url['path'], $path );
+		$tok = explode("/", $parsed_url['path']);
+		$tok = explode("_", $tok[2]);
```
Comment (mjangda) [inline includes/embeds/class-amp-soundcloud-embed.php]: Do we need the `_` explode here?

Comment (amedina) [inline includes/embeds/class-amp-soundcloud-embed.php]: My bad; not needed. Fixed.

**Comment by westonruter** [comment]
@amedina as part of

**Comment by westonruter** [comment]
@amedina as part of #806 I found the implementation here to not be complete. In particular, an oEmbed for SoundCloud looks like:

https://soundcloud.com/jack-villano-villano/mozart-requiem-in-d-minor

Not like:

https://api.soundcloud.com/tracks/89299804

Additionally, I found the `soundcloud` shortcode handling wasn't working as expected. For one thing, the Jetpack plugin only recognizes the `url` attribute whereas WordPress.com also apparently supports the `id` attribute.

So I've amended the PR #829 to fix up the SoundCloud handling: https://github.com/Automattic/amp-wp/pull/829/commits/0b729b1a728fbfd0faa9a7842ec757d1b8a5b4bd

**Pull request merged**